File: GoTvApps/GoTvViewerAndroid/app/src/main/java/com/p2pgotv/gotvstation/VxSha1Hash.java
//============================================================================
// Copyright (C) 2015 Brett R. Jones
// All Rights Reserved ( After My Death I Brett R. Jones Issue this source code to Public Domain )
//
// You may not modify or redistribute this code for any reason
// without written consent of Brett R. Jones
//
// This code is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
//
// brett1900@usa.com
// http://www.gotvp2p.net
//============================================================================
package com.p2pgotv.gotvstation;

public class VxSha1Hash 
{
	private static final int 	SHA1_HASH_LEN 					= 20;
	
	private byte				m_HashId[]						= new byte[ SHA1_HASH_LEN ];
	
	byte[]						getHashData()					{ return m_HashId; }
	
	//========================================================================
	VxSha1Hash()
	{
		for( int i = 0; i < SHA1_HASH_LEN; i++ )
		{
			m_HashId[i] = 0;
		}
	}
	
	//========================================================================
	void setHashData( byte[] hashData )
	{
		if( null != hashData )
		{
			for( int i = 0; i < SHA1_HASH_LEN; i++ )
			{
				m_HashId[i] = hashData[i];
			}			
		}
	}
	
	//========================================================================
	boolean	isHashValid()
	{
		for( int i = 0; i < SHA1_HASH_LEN; i++ )
		{
			if( 0 != m_HashId[i] )
			{
				return true;
			}
		}

		return false;		
	}
	
	//========================================================================
	boolean	isEqualTo( byte[] hashData )
	{
		if( null == hashData )
		{		
			return false;
		}
		
		for( int i = 0; i < SHA1_HASH_LEN; i++ )
		{
			if( hashData[i] != m_HashId[i] )
			{
				return false;
			}
		}	
		
		return true;
	}	
}

File: DependLibs/CoreLib/VxSha1Hash.cpp

//============================================================================
// Copyright (C) 2015 Brett R. Jones
// Issued to MIT style license by Brett R. Jones in 2017
//
// You may use, copy, modify, merge, publish, distribute, sub-license, and/or sell this software
// provided this Copyright is not modified or removed and is included all copies or substantial portions of the Software
//
// This code is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
//
// bjones.engineer@gmail.com
// http://www.gotvptop.net
//============================================================================
#include "config_corelib.h"

#include "VxSha1Hash.h"
#include "SHA1.h"

//============================================================================
VxSha1Hash::VxSha1Hash()
{
	memset( m_HashId, 0, sizeof( m_HashId ) );
}

//============================================================================
VxSha1Hash::VxSha1Hash( const VxSha1Hash &rhs )
{
	*this = rhs;
}

//============================================================================
VxSha1Hash& VxSha1Hash::operator =( const VxSha1Hash &rhs )
{
	if( this != &rhs )   
	{
		memcpy( m_HashId, rhs.m_HashId, sizeof( m_HashId ) );
	}

	return *this;
}

//============================================================================
bool VxSha1Hash::operator == ( const VxSha1Hash &a ) const
{
	return ( 0 == memcmp( m_HashId, a.m_HashId, sizeof( m_HashId ) ) ) ? true : false;
}

//============================================================================
bool VxSha1Hash::operator != ( const VxSha1Hash &a ) const
{
	return ! (*this == a);
}

//============================================================================
bool VxSha1Hash::isHashValid()	const
{
	uint32_t * longPtr = (uint32_t *)m_HashId;
	for( unsigned int i = 0; i < 4; i++ )
	{
		if( 0 != longPtr[0] )
		{
			return true;
		}
	}

	return false;
}

//============================================================================
bool VxSha1Hash::isEqualTo(  const uint8_t * hashData ) const
{
	bool isEqual = false;
	if( hashData )
	{
		isEqual = (0 == memcmp( hashData, m_HashId, sizeof( m_HashId ) ) );
	}

	return isEqual;
}


//============================================================================
void VxSha1Hash::setHashData( const uint8_t * data )			
{ 
	if( data )
	{
		memcpy( m_HashId, data, sizeof( m_HashId ) ); 
	}
	else
	{
		memset( m_HashId, 0, sizeof( m_HashId ) ); 
	}
}

//============================================================================
bool VxSha1Hash::generateHashFromFile( const char * fileName, VxThread * workThread )
{
	CSHA1 shaInstance;
	memset( m_HashId, 0, sizeof( m_HashId ) );
	bool result = shaInstance.HashFile( fileName, workThread );
	if( result )
	{
		shaInstance.Final();
		shaInstance.GetHash( m_HashId );
	}

	return result;
}
